### PR TITLE
Added room ut, updated patient ut

### PR DESCRIPTION
**Pull request by BaqWin**
#86 #90

### CODE REVIEW

File: src/tests/ut/room/room_test.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+class RoomTestFixture : public ::testing::Test
+{
+
+    void SetUp() override
+    {
+        Doctor::createDoctor("Jan", "Kowalski", "80062378532", Gender::Male);
+
+        Visit::createVisit(Clinic::getDoctors().at(0));
+        Visit::createVisit(Clinic::getDoctors().at(0));
```
Comment (fbtom) [inline src/tests/ut/room/room_test.hpp:16]: redundant call

File: src/tests/ut/patient/patient_test.hpp [hunk 2]
```diff
@@ -7,10 +7,16 @@
 
 class PatientTestFixture : public ::testing::Test
 {
+    void SetUp() override
+    {
+        Doctor::createDoctor("Gawel", "Pawel", "49042481257", Gender::Male);
+
+        Visit::createVisit(Clinic::getDoctors().at(0));
+        Visit::createVisit(Clinic::getDoctors().at(0));
```
Comment (fbtom) [inline src/tests/ut/patient/patient_test.hpp:15]: redundant call

File: src/app/room/room.cpp [hunk 3]
```diff
@@ -58,3 +62,14 @@ void Room::createRoom(const unsigned room_id, const std::vector<Treatment>& trea
     Room room{room_id, treatments, warehouse};
     Clinic::appendRoom(std::make_shared<Room>(room));
 }
+
+void Room::createRoom(const unsigned room_id, const std::shared_ptr<Warehouse>& warehouse)
+{
+    Room room{room_id, warehouse};
+    Clinic::appendRoom(std::make_shared<Room>(room));
+}
```
Comment (fbtom) [inline src/app/room/room.cpp:70]: Just curious if it could be shortened to 
`Clinic::appendRoom(std::make_shared<Room>({room_id,warehouse}));`?

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

File: src/tests/ut/room/room_test.hpp (hunk 1 above)
Comment (BaqWin) [reply src/tests/ut/room/room_test.hpp:16]: fixed

File: src/tests/ut/patient/patient_test.hpp (hunk 2 above)
Comment (BaqWin) [reply src/tests/ut/patient/patient_test.hpp:15]: fixed

File: src/app/room/room.cpp (hunk 3 above)
Comment (BaqWin) [reply src/app/room/room.cpp:70]: Looks like it does, need to implement it in every create function

**Review by fbtom — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @BaqWin for making course materials better!

**Pull request merged**